### ISSUE TITLE
Communication error on melonDS core

### ISSUE THREAD

**Issue by gabe-otter** [Swordfish90/Lemuroid#151]
I'm running into an error when loading a pokemon platinum ROM, for example. It seems to work fine on the Desmume core, but I switched to MelonDS because it looked better. The save states work fine in loading and saving (which is my current work around), but for some reason loading the game through the actual save file (clicking restart on the ROM, and loading through the platinum menu) gives me the "communication error" attached, followed by a white screen that never goes away. I can then only restart the game and get the same issue. Loading a save state seems to work fine, though. Deleting my saves and starting a brand new file also gave me the same communication error. 

I am running the 1.8.0 version on a OnePlus 7t, if that is important. 

![Screenshot_20201222-172909](https://user-images.githubusercontent.com/76503567/102950737-13958200-4480-11eb-9c53-2e692662df46.jpg)

**Comment by Swordfish90** [comment]
Thank you for reporting this. Are you able to reproduce the issue only with Pokemon Platinum or is the issue widespread with your setup?

**Comment by gabe-otter** [comment]
Hi, I've just tried with Pokemon White version and was not able to reproduce the communication error, it seems to be only with the Platinum. I haven't tried other games yet but it seems like the problem might be coming from some nintendo wi-fi settings. I have not enabled them in either game (in fact, I'm not even able to load the menus screen before I get the error in Platinum). I'll see if I can find other games to reproduce the error with. 

I've also downloaded MelonDS for Desktop on Windows, transferred my Platinum save file from the Lemuroid files location with the same ROM and I can actually load everything fine, so I doubt it is the ROM or a corrupt save file. And again the save states work on Lemuroid just fine.

**Comment by Ashiix** [comment]
I have some more info about this, it can be fixed by disabling audio syncing in emulators that support this option. If you can add the option to disable it in Lemuroid it can be fixed very easily.

**Comment by Ashiix** [comment]
If, at some point, a config file is added, adding "audio_sync = false" to it will make it work.

**Comment by shabbirmahmud** [comment]
I get this communication error in every ds games like pokemon soul silver,white 2 etc.My core is melonds.I dont get this error on DeSmuMe.
![Screenshot_2021-03-04-08-39-09-24_21f12b1316663fbd369c35432dda51b0](https://user-images.githubusercontent.com/47573399/109903015-6f248d00-7cc5-11eb-9c7e-cade2ae9efca.jpg)
![Screenshot_2021-03-04-08-39-43-60_21f12b1316663fbd369c35432dda51b0](https://user-images.githubusercontent.com/47573399/109903030-7481d780-7cc5-11eb-80af-3535f50f468e.jpg)

**Comment by shabbirmahmud** [comment]
Duplicate of #184

**Comment by pokegamer5547** [comment]
Im having same issue with all of my DS pokemon games when using melon ds core

**Comment by D4NH** [comment]
I don't know if this will help, but I got the game running again by renaming the rom.

**Comment by Dasnap** [comment]
I'll add that I am also having this issue with Pokémon Soul Silver. The very first play session will be fine, but the first save, quit, and relaunch causes the communication error.

**Comment by Dasnap** [comment]
I've tested with HeartGold also and the same occurs.

**Comment by shabbirmahmud** [comment]
I get this all games which has wifi support and try to connect to the wifi after entering the game.

**Comment by max-las** [comment]
This is because firmware and/or BIOS are missing. The same issue occurs in Retroarch when you don't set up BIOS and firmware, and it's gone when they are present.

**Comment by Caiuz89** [comment]
I'm having the same issue with pokemon soul silver, so im using the save states to continue.

**Comment by Larichan** [comment]
@max-las is right. I was having the same problem, and although my firmware and BIOS archives are in the right place, they weren't named correctly. Renamed them and worked properly on Pokémon Platinum.

**Comment by Jevitate** [comment]
How would you go about changing the file names on Android phone to fix this error for the pokemon games?

**Comment by Dasnap** [comment]
I oddly enough can't seem to get firmware.bin to be picked up on both my A52 and Chromecast, even though it is in an available directory.
![Screenshot_20220114-133155_Lemuroid](https://user-images.githubusercontent.com/16799832/149523761-fd3b46b5-528e-4d5d-85d6-e96917fe3863.jpg)
![Screenshot_20220114-133231_My Files](https://user-images.githubusercontent.com/16799832/149523780-bfb4641a-d278-4b2a-8063-5a126ca71943.jpg)

**Comment by MyFilippo94** [comment]
Having the same issues, I also have this error loading saves with HeartGold, and can't seem to get firmware.bin recognized by Lemuroid.

**Comment by Rodeeo** [comment]
Same issue remains, Heartgold using melonds. Beat the elite 4 and get the game over screen and cant go back into the save file.

**Comment by mia-cx** [comment]
I encountered this error in platinum only after entering the underground on desmume, i read that it needs true wifi in platinum and only desmume supports that so far.

after saving my game in desmume AFTER getting items from the underground and having put them in my bag, melonds is unable to load the save file from the rom

**Comment by kingsick96** [comment]
So Iv been digging around all day unfortunately I made the mistake of a saving in game and would get the communication error using the emulator on RetroArch Plus for android. I found out that you haft to delete the save file in order to solve this. I opened Google files and searched for Pokemon heartgold save and found the 300kb save file and deleted it. You won't be able to recover any game time but you can now play!! I hope this helps everyone and please spread the word

**Comment by Geezgus** [comment]
There's no need for deleting your data AFAIK, but when using melonDS, make sure to provide firmware files (usually called `nds_firmware.bin` or `firmware.bin`). that's basically it. just put this file wherever your roms are and choose it on BIOS settings. should be good to go

**Comment by ero57** [comment]
Solution for In-Game Save / Communication Error on melonDS

I found a fix for the communication error when saving/loading Pokémon games (Setup: melonDS on Arch Linux):

1. Go to Config -> Emu settings -> DS Mode.
2. Link your BIOS and Firmware files.
3. Uncheck the box for "Use external BIOS/firmware files".
4. Restart the emulator and load your save file normally.

After unchecking that box, the save files loaded perfectly without errors.
